src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java

```diff
@@ -122,6 +142,41 @@ public QuizExerciseResource(QuizExerciseRepository quizExerciseRepository, Parti
         //notify clients via websocket about the release state of the statistics.
         statisticService.releaseStatistic(quizExercise, quizExercise.getQuizPointStatistic().isReleased());
 
+        //change existing results if an answer or and question was deleted
+        for (Result result : resultRepository.findByParticipationExerciseIdOrderByCompletionDateAsc(quizExercise.getId())) {
+
+            Set<SubmittedAnswer> submittedAnswersToDelete = new HashSet<>();
+
+            for (SubmittedAnswer submittedAnswer : ((QuizSubmission) result.getSubmission()).getSubmittedAnswers()) {
+                if (submittedAnswer instanceof MultipleChoiceSubmittedAnswer) {
+                    // Delete all references to question an answers if the question was deleted
+                    if (!quizExercise.getQuestions().contains(submittedAnswer.getQuestion())) {
+                        submittedAnswer.setQuestion(null);
+                        ((MultipleChoiceSubmittedAnswer) submittedAnswer).setSelectedOptions(null);
+                        submittedAnswersToDelete.add(submittedAnswer);
+                    } else {
+                        // find same question in quizExercise
+                        for (Question question : quizExercise.getQuestions()) {
+                            if (question.getId().equals(submittedAnswer.getQuestion().getId())) {
+                                // Check if an answerOption was deleted and delete reference to in selectedOptions
+                                Set<AnswerOption> selectedOptionsToDelete = new HashSet<>();
+                                for (AnswerOption answerOption : ((MultipleChoiceSubmittedAnswer) submittedAnswer).getSelectedOptions()) {
+                                    if (!((MultipleChoiceQuestion) question).getAnswerOptions().contains(answerOption)) {
+                                        selectedOptionsToDelete.add(answerOption);
+                                    }
+                                }
+                                ((MultipleChoiceSubmittedAnswer) submittedAnswer).getSelectedOptions().removeAll(selectedOptionsToDelete);
+
+                            }
+                        }
+                    }
+                }
+                // TO-DO DragAndDrop Question
+            }
+            ((QuizSubmission) result.getSubmission()).getSubmittedAnswers().removeAll(submittedAnswersToDelete);
+            quizSubmissionRepository.save((QuizSubmission) result.getSubmission());
+        }
+
```

Comment:
I don't understand why we need this code in the normal PUT? Isn't that impossible, because there are no results yet when the quiz has not started?!?

Reply:
@MoritzIssig can you please answer this question?

Reply:
You are right. In the normal PUT this is impossible, so I moved it to the reEvaluateQuizExercise-PUT.